chore/SOF-7155: Remove SCSS styling and migrate to MUI

timurbazhirov opened the pull request.

timurbazhirov committed:
chore: remove scss files

timurbazhirov committed:
chore: remove main.scss content except for wave.js height

timurbazhirov committed:
chore: fix menu buttons

timurbazhirov committed:
chore: move border to editor selection info

timurbazhirov committed:
chore: add AppBar, remove SourceEditor

timurbazhirov committed:
chore: newline

timurbazhirov committed:
chore: fix ItemsList items styling

timurbazhirov committed:
chore: updatge the height of the footer and related

timurbazhirov committed:
chore: fix tests

timurbazhirov committed:
style: menu button colors

timurbazhirov committed:
style: border color

timurbazhirov committed:
responsive height: for main component

timurbazhirov committed:
fix: edited index to -1 initially

Nothing is being edited on first load

timurbazhirov committed:
feat: attempt making menu items dense

For some reason this doesn't work, only when ScopedCSSBaseLine is put outside ThemeProvider

timurbazhirov committed:
feat: add support for a "logo" icon in header appbar

timurbazhirov committed:
chore: remove bootstrap

timurbazhirov committed:
chore: add typography to accodion summaries

timurbazhirov committed:
chore: add ScopedCssBaseline to color scrollbars etc. in dark mode

Also add icon to AppBar

timurbazhirov committed:
style: ItemsList items, padding

timurbazhirov committed:
chore: minor cleanup

timurbazhirov committed:
style: height for crystal basis accordion

timurbazhirov committed:
chore: update package.json - remove bootstrap

timurbazhirov committed:
chore: update package-lock

timurbazhirov committed:
style: increase code mirror font size a bit

timurbazhirov committed:
chore: hide some menu items with ternary logic

timurbazhirov committed:
chore: update wave.js version

timurbazhirov committed:
chore: rebuild package-lock.json file from package.json

VsevolodX commented on src/components/header_menu/HeaderMenuToolbar.jsx:
```diff
@@ -491,11 +499,14 @@ HeaderMenuToolbar.propTypes = {
     closeImportModal: PropTypes.func.isRequired,
     toggleFullscreen: PropTypes.func.isRequired,
```
Fullscreen option doesn't do anything:
![image](https://github.com/Exabyte-io/materials-designer/assets/79542055/19463f03-4cd8-4008-964c-7e469b5ce52b)

Do we remove it completely?

timurbazhirov committed:
negative-feature: remove fullscreen trigger

plus: clean up xl breakpoints

We don't need fullscreen when parts of the interface can be hidden

timurbazhirov committed:
chore: fix tests selector

timurbazhirov committed:
chore: fix tests selector x 2

timurbazhirov committed:
chore: fix tests selector x 3

timurbazhirov committed:
chore: newline

VsevolodX reviewed (approved):
Verified the looks locally
